contracts/src/v0.8/dev/functions/FunctionsOracle.sol

```diff
@@ -34,7 +40,13 @@ contract FunctionsOracle is FunctionsOracleInterface, OCR2Base, AuthorizedOrigin
   FunctionsBillingRegistryInterface private s_registry;
   mapping(address => bytes) private s_nodePublicKeys;
 
-  constructor() OCR2Base(true) {}
+  /**
+   * @dev Initializes the contract.
+   */
+  function initialize() public initializer {
```

Comment:
Would it be possible for a malicious user to call this from its own contract constructor in between of this contract deployment and a proxy switch?

Reply:
I like your thinking.

It won't be possible. The initialize functions are [only callable once](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/utils/Initializable.sol#L86).
Any contract upgrades that introduce a new `initialize` function are called in within their own deployment and also only callable once.

Reply:
Right, I know it can be called once. Now imagine hackers know our deployer/signer address (easy to guess from Etherscan). They could be tracking all contract deployments initiated by the address and for all newly deployed contracts they can call initialize(). Then your upgrade flow would break. You would have to repeat deployment, etc. This is a sort of DoS attack, as long as an attacker knows our deploying address. Would it make sense to add some modifier like `onlyOwner` to prevent this?